### DISCUSSION TITLE
How to override "Email Verification" action token

### DISCUSSION THREAD

**Discussion by shahlokesh02** [keycloak/keycloak#19202] [Q&A]
I am trying to increase the timeout for the `Email Verification` (under `realms --> Tokens --> Access tokens`) to a value greater than default `User-Initiated Action Lifespan `for the realm but this does not work. Email verification emails always contain the default timeout value rather than new override value of token.
Does anyone had also similar issue and how to resolve it?

**Comment by beg1shev** [comment]
@shahlokesh02  Did you find out how to solve the issue?

**Comment by fidgi** [comment]
I faced the same issue when trying to specified a value for the `Forgot password` timeout .

**Reply by fidgi** [reply]
May be this is because , the lifespan override per action type doesn't apply to the user-intiated action but only to the admin-initiated action ?

**Comment by ayumi-aikawa** [comment]
I faced it and it may be bug.
Admin console send the `Email Verification` as `actionTokenGeneratedByUserLifespan-verify-email` and the param is registered to DB  for realm_attribute table.
But the server side try to use the param as `actionTokenGeneratedByUserLifespan.verify-email` (separate character is `.` instead of `-`) then Keycloak can't pick the attribute as action token lifespan.
https://github.com/keycloak/keycloak/blob/004805e21d1b5cb20afc4307fd1c55942c3f19c7/model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java#L640-L646

`Forgot password` has same problem. admin console send it as `actionTokenGeneratedByUserLifespan-reset-credeltials`.

Note:
I'm using ver 21.0.4 and I didn't check latest admin console. so there is possibility the problem already resolved in latest version.

**Comment by ayumi-aikawa** [comment]
This problem resolved at v23.0.5.
https://github.com/keycloak/keycloak/pull/25996/files
